src/adapters/suggest_sources.js

```diff
@@ -0,0 +1,55 @@
+import PoiStore from './poi/poi_store';
+import BragiPoi from './poi/bragi_poi';
+import CategoryService from './category_service';
+
+// @TODO: Improvement: don't access directly to window.map
+function getFocus(focusMinZoom) {
+  const zoom = window.map && window.map.mb && window.map.mb.getZoom();
+  if (zoom >= focusMinZoom) {
+    const { lat, lon } = window.map.mb.getCenter();
+    return { lat, lon, zoom };
+  }
+  return {};
+}
+
+export function suggestResults(term, { withCategories, useFocus, focusMinZoom = 11 } = {}) {
+  let geocoderPromise;
+  let promise;
+  if (term === '') {
```

Comment:
I think we can improve readability further by opting for early returns

Reply:
I thought about it but as the current implementation requires attaching an 'abort' function to the returned promise in any case, I don't think that's possible.
BTW, I expect we'll be able to clean more things once everything is plugged to your new implementation.